nav2_bringup/launch/pose_persistent_launch.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2024 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Launch file for persistent pose saver/restorer node."""
+
+import os
+from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+from launch_ros.actions import Node
+
+def generate_launch_description():
+    bringup_dir = get_package_share_directory('nav2_bringup')
+
+    # Declare Launch Arguments
+    declare_auto_start_saving_cmd = DeclareLaunchArgument(
+        'auto_start_saving', default_value='true',
+        description='Whether to automatically start saving poses')
+
+    declare_auto_restore_pose_cmd = DeclareLaunchArgument(
+        'auto_restore_pose', default_value='true',
+        description='Whether to automatically restore pose on startup')
+
+    declare_pose_file_path_cmd = DeclareLaunchArgument(
+    'pose_file_path', 
+    default_value=os.path.join(
```

Comment:
The filepath should be something set in the configuration of the node and should use some directory that the user sets up (`/home/`, `/tmp`, etc). Using a file in the shared install space of a package is very odd and we shouldn't be modifying the source's install. It should be in another location on the filesystem not related to the source / installed software to persist updates.